### PR TITLE
Add openOCD format config generation for IOMUX

### PR DESCRIPTION
**Pull request by kowalewskijan**
Signed-off-by: Jan Kowalewski <jkowalewski@antmicro.com>

### CODE REVIEW

**Comment by kowalewskijan** [comment]
I can see that there is target named e.g. `counter-ql-chandalar_jlink`, so I wonder if the default format should be either openocd or jlink? Or should I also change the target recipe in CMake somewhere?

File: quicklogic/common/utils/eos_s3_iomux_config.py [hunk 1]
```diff
@@ -239,10 +246,17 @@ def main():
...
     print("")
-    for adr in sorted(iomux_regs.keys()):
-        print("w4 0x{:08X} 0x{:08X}".format(adr, iomux_regs[adr]))
+    if args.output_format == "openocd":
+        # Output openOCD process
+        print('proc setup_iomux {} {')
+        for adr in sorted(iomux_regs.keys()):
+            print("    mww 0x{:08X} 0x{:08X}".format(adr, iomux_regs[adr]))
+        print('}')
+    else:
```
Comment (mkurc-ant) [inline quicklogic/common/utils/eos_s3_iomux_config.py:256]: I'd use here `elif args.output_format == "jlink"` and then later print some error message under the final `else`

**Comment by mkurc-ant** [comment]
@kowalewskijan So now since we can use OpenOCD it can be added to the flow. There is a CMake function called `add_jlink_output` defined in `quicklogic/common/cmake/quicklogic_jlink.cmake`. You may add something identical for OpenOCD.

There is also some other CMake code that generates targets for programming boards. It is used in xc7. Maybe that could also be adapted here. I haven't used those so I can't tell.

**Comment by kowalewskijan** [comment]
I added the openocd stage to quicklogic tests. I managed to get both `top_iomux.jlink` and `top_iomux.openocd` by calling `<test>-ql-<board>_jlink` and `<test>-ql-<board>_openocd` respectively. I tried to use bitstream generated from e.g. `btn_counter` but it appeared that for both `jlink` and `openocd` I always get just:
```
<cmd> 0x40004D80 0x00000000
<cmd> 0x40004D84 0x00000000
```
I believe that EBLIF and PCF are not parsed properly for some reason.

**Comment by mkurc-ant** [comment]
@kowalewskijan So the thing is that information about chip pad to correct FBIO index is no longer present anywhere. PCF constraints used to express pins as `FBIO_n` but now they refer directly to chip pads. The scripts work fine but they always return "empty" IOMUX settings.

That is something that needs to be fixed in the future provided that we still want automatic IOMUX configuration for FPGA designs.

**Pull request merged**